### DISCUSSION TITLE
How to change the default host and port for HttpServerLauncher ?

### DISCUSSION THREAD

**Discussion by Soumyajit-Dutta** [activej/activej#288] [Q&A]
Hi

  I am trying to create a httpserver which will be accept a post request. I have created a http-server.properties file and added the
`http.listenAddresses=152.1.1.1:80` . Yet when I am running my code it is defaulting to the default host and port which is localhost and 8080.

My code is below.

```
import io.activej.config.Config;
import io.activej.http.*;
import io.activej.inject.annotation.Provides;
import io.activej.launcher.Launcher;
import io.activej.launchers.http.HttpServerLauncher;
import java.util.concurrent.Executor;
import java.util.concurrent.Executors;
import static WebhookBackend.extractFromResponse;
import static io.activej.http.HttpMethod.POST;


public final class WebServer extends HttpServerLauncher {
    @Provides
    Executor executor() {
        return Executors.newCachedThreadPool();
    }
    @Provides
    AsyncServlet servlet(Executor executor) {
        return RoutingServlet.create()
                .map(POST, "/my_webhook", request -> request.loadBody()
                        .map($ -> {
                            String quoteResponse = extractFromResponse(String.valueOf(request.getBody()),"text");
                            System.out.println(quoteResponse);
                            return HttpResponse.ok200()
                                    .withHtml("Webhook Received in Java");
                        }));
    }


    public static void main(String[] args) throws Exception {

//        Config config = Config.create()
//            .with("http.listenAddresses", "152.1.1.1:80");


        Launcher launcher = new WebServer();
        launcher.launch(args);
    }
    }

```

**Comment by eduard-vasinskyi** [comment]
@Soumyajit-Dutta

Please, validate that your `.properties` file is on the classpath. You can check that by adding `Config.ofClassPathProperties("http-server.properties");` line at the start of the `main()` method. If that line fails - properties file is not on the classpath. 

Alternatively, you can write your own HTTP server launcher from scratch (as HttpServerLauncher is an example class). That way you can control how properties are passed to the application and HTTP server. Or you can use HttpServer directly without DI (as in this [example](https://github.com/activej/activej/blob/505a3439ee955f60c56a104cd5071afc31db901b/examples/core/http/src/main/java/HelloWorldExample.java))
